lib/chef/provisioning/docker_driver/docker_transport.rb

```diff
@@ -179,11 +179,12 @@ def upload_file(local_path, path)
     def make_url_available_to_remote(url)
       # The host is already open to the container.  Just find out its address and return it!
       uri = URI(url)
+      uri.scheme = 'http' if 'chefzero' == uri.scheme && uri.host == 'localhost'
```

Comment:
I should have already fixed this in chef-provisioning core in https://github.com/chef/chef-provisioning/pull/337, which is available in chef-provisioning 1.2